hs_composite_resource/models.py

```diff
@@ -0,0 +1,180 @@
+import os
+
+from mezzanine.pages.page_processors import processor_for
+
+from hs_core.models import BaseResource, ResourceManager, resource_processor
+
+from hs_file_types.models import GenericLogicalFile
+
+
+class CompositeResource(BaseResource):
+    objects = ResourceManager("CompositeResource")
+
+    class Meta:
+        verbose_name = 'Composite Resource'
+        proxy = True
+
+    @property
+    def can_be_public_or_discoverable(self):
+        # resource level metadata check
+        if not super(CompositeResource, self).can_be_public_or_discoverable:
+            return False
+
+        # filetype level metadata check
+        for lf in self.logical_files:
+            if not lf.metadata.has_all_required_elements():
+                return False
+
+        return True
+
+    def set_default_logical_file(self):
+        """sets an instance of GenericLogicalFile to any resource file objects of this instance
+        of the resource that is not already associated with a logical file. """
+
+        for res_file in self.files.all():
+            if not res_file.has_logical_file:
+                logical_file = GenericLogicalFile.create()
```

Comment:
The reason for not using key/value pairs for logical file type metadata is to use existing code as much as possible between raster resource type and raster logical file type. If I remember correctly this was also your approach as noted in Hydroshare weekly meeting notes (google doc). 

I would prefer to use key/value for metadata everywhere and I can try this approach for implementing netcdf file type - which is next in pipeline for me.